### PR TITLE
add icon to show and hide balance

### LINKED ISSUE #589: button to show/hide wallet balance
on the overview page, we could look at possibility of adding an icon beside the balance(s), to show/hide the wallet balance, this is desirable because it enhances the users privacy

### PR DESCRIPTION
**Pull request by dreacot**
Resolves #589

This PR adds an icon to hide and show the balance on the overview page

**Screenshots**
| | |
|-|-|

### CODE REVIEW

File: app/src/main/java/com/dcrandroid/fragments/OverviewFragment.kt [hunk 1]
```diff
@@ -154,7 +182,33 @@ class OverviewFragment : BaseFragment(), ViewTreeObserver.OnScrollChangedListene
...
+        ivConcealReveal.setOnClickListener {
+            if (isBalanceHidden) {
+                multiWallet!!.setBoolConfigValueForKey(Constants.SHOW_HIDE_BALANCE, false)
+                isBalanceHidden = false
+                loadBalance()
+            } else {
+                multiWallet!!.setBoolConfigValueForKey(Constants.SHOW_HIDE_BALANCE, true)
+                isBalanceHidden = true
+                loadBalance()
+            }
```
Comment (JustinBeBoy) [inline app/src/main/java/com/dcrandroid/fragments/OverviewFragment.kt:194]: You can replace with 
```suggestion
            isBalanceHidden = !isBalanceHidden
            multiWallet?.setBoolConfigValueForKey(Constants.SHOW_HIDE_BALANCE, isBalanceHidden)
            loadBalance()
```

File: app/src/main/java/com/dcrandroid/fragments/OverviewFragment.kt [hunk 2]
```diff
@@ -154,7 +182,33 @@ class OverviewFragment : BaseFragment(), ViewTreeObserver.OnScrollChangedListene
...
+        toolbar_right_icon.setOnClickListener {
+            if (isBalanceHidden) {
+                multiWallet!!.setBoolConfigValueForKey(Constants.SHOW_HIDE_BALANCE, false)
+                isBalanceHidden = false
+                loadBalance()
+            } else {
+                multiWallet!!.setBoolConfigValueForKey(Constants.SHOW_HIDE_BALANCE, true)
+                isBalanceHidden = true
+                loadBalance()
+            }
```
Comment (JustinBeBoy) [inline app/src/main/java/com/dcrandroid/fragments/OverviewFragment.kt:206]: Same as above

File: app/src/main/java/com/dcrandroid/fragments/OverviewFragment.kt [hunk 3]
```diff
@@ -89,6 +104,15 @@ class OverviewFragment : BaseFragment(), ViewTreeObserver.OnScrollChangedListene
         usdBalanceTextView = view.findViewById(R.id.tv_visible_usd_wallet_balance)
         noTransactionsTextView = view.findViewById(R.id.tv_no_transactions)
         transactionsLayout = view.findViewById(R.id.transactions_view)
+        ivConcealReveal = view.findViewById(R.id.iv_conceal_reveal)
+
+        val toolbar: Toolbar = view.findViewById(R.id.toolbar)
+        toolbarTitle = toolbar.findViewById(R.id.toolbar_title)
+        toolbarSubtitle = view.findViewById(R.id.toolbar_subtitle)
+//        val imv: ImageView = toolbar.findViewById(R.id.toolbar_right_icon)
+//        imv.visibility = View.VISIBLE
```
Comment (JustinBeBoy) [inline app/src/main/java/com/dcrandroid/fragments/OverviewFragment.kt:113]: Remove

File: app/src/main/java/com/dcrandroid/fragments/OverviewFragment.kt [hunk 4]
```diff
@@ -106,13 +130,17 @@ class OverviewFragment : BaseFragment(), ViewTreeObserver.OnScrollChangedListene
         setToolbarTitle(R.string.overview, false)
         scrollView.viewTreeObserver.addOnScrollChangedListener(this)
 
+        isBalanceHidden = multiWallet!!.readBoolConfigValueForKey(Constants.SHOW_HIDE_BALANCE, false)
```
Comment (JustinBeBoy) [inline app/src/main/java/com/dcrandroid/fragments/OverviewFragment.kt:133]: We should not use `!!`

File: app/src/main/java/com/dcrandroid/fragments/OverviewFragment.kt [hunk 5]
```diff
@@ -391,15 +501,17 @@ class OverviewFragment : BaseFragment(), ViewTreeObserver.OnScrollChangedListene
...
+                    getString(
+                            R.string.usd_symbol_format,
+                            CurrencyUtil.dcrToFormattedUSD(exchangeDecimal, totalBalanceCoin, 2)
+                    ), 0
             )
 
             GlobalScope.launch(Dispatchers.Main) {
-                usdBalanceTextView.text = formattedUSD
-                usdBalanceTextView.show()
+                if (!isBalanceHidden) {
```
Comment (JustinBeBoy) [inline app/src/main/java/com/dcrandroid/fragments/OverviewFragment.kt:511]: We can use `MainScope().launch { }` to replace `GlobalScope`

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -676,6 +676,8 @@
     <string name="no_votes_percent">No: %1$d (%2$.2f%%)</string>
     <string name="yes_no_votes_percent">Yes: %1$d (%2$.2f%%)  No: %3$d (%4$.2f%%)</string>
     <string name="proposal_name_author">%1$s by %2$s</string>
+    <string name="show_balance">Show Balance</string>
+    <string name="hide_balance">Hide Balance</string>
```
Comment (JustinBeBoy) [inline app/src/main/res/values/strings.xml:680]: Move its out of `Politeia` tag

**Review by JustinBeBoy — CHANGES_REQUESTED** [review]
I have some suggestion changes

File: app/src/main/java/com/dcrandroid/fragments/OverviewFragment.kt (hunk 4 above)
Comment (dreacot) [reply app/src/main/java/com/dcrandroid/fragments/OverviewFragment.kt:133]: the use of `!!` is important and acceptable for certain use cases, for example you need the app to crash when a vital package or resource isn't available

`multiwallet` is part of the foundation of our apps, if for some reason multiwallet tends to be nil, there are countless errors that would occur as a result of that, and for that scenario, it's best for the app to panic and shutdown

File: app/src/main/java/com/dcrandroid/fragments/OverviewFragment.kt (hunk 5 above)
Comment (dreacot) [reply app/src/main/java/com/dcrandroid/fragments/OverviewFragment.kt:496]: what's the advantage of `MainScope().launch { }` in this case?

**Review by yaroyazeed — APPROVED** [review]

**Pull request merged**